protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,387 @@
+    DistributionParams private distributionParams;
+    /// @notice Vesting factor determines rate of mint
+    // This represents the vesting factor
+    // If mint every 1 second instead of 1 week, vestingFactor == 35 / 604,800. It is possible but unlikely vesting factor is changed in future
+    VestingFactor private vestingFactor;
+
+    constructor(
+        InitArgs memory _initArgs
+    ) OFT(_initArgs.name, _initArgs.symbol, _initArgs.layerZeroEndpoint, _initArgs.executor) Ownable(_initArgs.executor){
+       _mintChainId = _initArgs.mintChainId;
+       _mintChainLzEid = _initArgs.mintChainLzEid;
+    }
+
+    /**
+     * @notice Set staking proxy contract address
+     * @param _staking Staking proxy contract
+     */
+    function setStaking(address _staking) external override onlyOwner {
+        if (_staking == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        staking = ITempleGoldStaking(_staking);
+        emit StakingSet(_staking);
+    }
+
+    /**
+     * @notice Set dai gold auction contract address
+     * @param _daiGoldAuction  contract address
+     */
+    function setDaiGoldAuction(address _daiGoldAuction) external override onlyOwner {
+        if (_daiGoldAuction == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        daiGoldAuction = IDaiGoldAuction(_daiGoldAuction);
+        emit DaiGoldAuctionSet(_daiGoldAuction);
+    }
+
+    /**
+     * @notice Set team gnosis address
+     * @param _gnosis Team gnosis address
+     */
+    function setTeamGnosis(address _gnosis) external override onlyOwner {
+        if (_gnosis == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        teamGnosis = _gnosis;
+        emit TeamGnosisSet(_gnosis);
+    }
+
+    /**
+     * @notice Whitelist an address to allow transfer of Temple Gold to or from
+     * @param _contract Contract address to whitelist
+     * @param _whitelist Boolean whitelist state
+     */
+    function authorizeContract(address _contract, bool _whitelist) external override onlyOwner {
+        if (_contract == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_contract] = _whitelist;
+        emit ContractAuthorizationSet(_contract, _whitelist);
+    } 
+
+    /**
+     * @notice Set distribution percentages of newly minted Temple Gold
+     * @param _params Distribution parameters
+     */
+    function setDistributionParams(DistributionParams calldata _params) external override onlyOwner {
+        if (_params.staking + _params.gnosis + _params.daiGoldAuction != DISTRIBUTION_DIVISOR) { revert ITempleGold.InvalidTotalShare(); }
+        distributionParams = _params;
+        emit DistributionParamsSet(_params.staking, _params.daiGoldAuction, _params.gnosis);
+    }
+
+    /**
+     * @notice Set vesting factor
+     * @param _factor Vesting factor
+     */
+    function setVestingFactor(VestingFactor calldata _factor) external override onlyOwner {
```

Comment:
```suggestion
    function setVestingFactor(VestingFactor calldata _factor) external override onlyOwner onlyArbitrum {
```

Although you can put the `onlyArbitrum` modifier in multiple functions that are only related to the minting logic (`setStaking()`, `setDaiGoldAuction()`, `setTeamGnosis()`, ...) I'm proposing this one in particular because of the following:

The view functions related to minting (`getMintAmount()`, `canDistrbute()` ) rely on `_getMintAmount()`. In non-arbitrum chains, this function should always return 0, but this is only the case, as long as `_lastMintTimestamp == 0`. 

And it turns out, `_lastMintTimestamp` is set to `block.timestamp` when calling `setVestingFactor()` for the first time, which doesn't trigger mint. 

So if `setVestingFactor()` is called by accident in a non-arbitrum chain, `_lastMintTimestamp > 0`, and then :
- `getMintAmount() > 0`
- `_canDistribute() = true`

tiny effect, but also tiny effort to fix it

Reply:
Good point